netcdftime/_netcdftime.pyx

```diff
@@ -1151,8 +1156,209 @@ def time2index(times, nctime, calendar=None, select='exact'):
     return _toscalar(index)
 
 
-def _toscalar(a):
+cdef _toscalar(a):
     if a.shape in [(), (1,)]:
         return a.item()
     else:
         return a
+
+
+cdef class datetime(object):
+
+    """
+Phony datetime object which mimics the python datetime object,
+but allows for dates that don't exist in the proleptic gregorian calendar.
+Doesn't do timedelta operations, doesn't overload + and -.
+
+Has strftime, timetuple and __repr__ methods.  The format
+of the string produced by __repr__ is controlled by self.format
+(default %Y-%m-%d %H:%M:%S). Does support comparisons with other
+phony datetime and with datetime.datetime objects.
+
+Instance variables are year,month,day,hour,minute,second,dayofwk,dayofyr
+and format.
+    """
+    cdef readonly int year, month, day, hour, minute, dayofwk, dayofyr
+    cdef readonly int second, microsecond
+    cdef str _calendar
+    cdef object _utime
+
+    def __init__(self, year, month, day, hour=0, minute=0, second=0,
+                 microsecond=0,dayofwk=-1, dayofyr=1, calendar="standard"):
+        """dayofyr set to 1 by default - otherwise time.strftime will complain"""
+
+        self.year = year
+        self.month = month
+        self.day = day
+        self.hour = hour
+        self.minute = minute
+        self.dayofwk = dayofwk
+        self.dayofyr = dayofyr
+        self.second = second
+        self.microsecond = microsecond
+        self._calendar = calendar
+        # Can't initialize _utime here: utime creates a datetime instance (origin).
+        # This avoids an unbounded recursion.
+        self._utime = None
+
+    @property
+    def format(self):
+        return '%Y-%m-%d %H:%M:%S'
+
+    def strftime(self, format=None):
+        if format is None:
+            format = self.format
+        return _strftime(self, format)
+
+    def replace(self, **kwargs):
+        "Return datetime with new specified fields."
+        args = {"year": self.year,
+                "month": self.month,
+                "day": self.day,
+                "hour": self.hour,
+                "minute": self.minute,
+                "second": self.second,
+                "microsecond": self.microsecond,
+                "dayofwk": self.dayofwk,
+                "dayofyr": self.dayofyr,
+                "calendar": self.calendar}
+
+        for name, value in kwargs.items():
+            args[name] = value
+
+        return datetime(**args)
+
+    def timetuple(self):
+        return (self.year, self.month, self.day, self.hour,
+                self.minute, self.second, self.dayofwk, self.dayofyr, -1)
+
+    def _to_real_datetime(self):
+        return real_datetime(self.year, self.month, self.day,
+                             self.hour, self.minute, self.second,
+                             self.microsecond)
+
+    def __repr__(self):
+        return self.strftime(self.format)
```

Comment:
maybe better to change the repr to include full type information, including the calendar?